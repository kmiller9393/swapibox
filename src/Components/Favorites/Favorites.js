import React, { Component } from 'react';

export default class Favorites extends Component {
  constructor() {
    super();
    this.state = {};
  }

  render() {
    return <div className="favorites">FAVORITES</div>;
  }
}
