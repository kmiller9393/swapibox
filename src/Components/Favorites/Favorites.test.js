import React from 'react';
import { shallow, mount } from 'enzyme';

import { Favorites } from './Favorites';

describe('Favorites', () => {
  let wrapper;
  let mockSetContainerView;
  let mockNumberOfFavorites;
  let mockEvent = {target: { title: 'favorites'}, preventDefault: jest.fn()};

  beforeEach(() => {
    mockSetContainerView = jest.fn();
    mockNumberOfFavorites = 1;
    wrapper = shallow(<Favorites
      setContainerView={mockSetContainerView}
      numberOfFavorites={mockNumberOfFavorites}
    />);
  });

  it('should call handleClick when btn is clicked', () => {
    
    wrapper.find('a').simulate('click');
    expect(mockSetContainerView).toHaveBeenCalledWith('favorites');

  });
  
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});