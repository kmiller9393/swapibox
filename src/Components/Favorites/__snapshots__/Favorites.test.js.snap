// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    numberOfFavorites={1}
    setContainerView={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <a
        class="superBtn"
        onClick={[Function]}
>
        Favorites:
        <span>
                 
                1
        </span>
</a>,
      "className": "favorites",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Favorites:",
          <span>
             
            1
</span>,
        ],
        "class": "superBtn",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        "Favorites:",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              1,
            ],
          },
          "ref": null,
          "rendered": Array [
            " ",
            1,
          ],
          "type": "span",
        },
      ],
      "type": "a",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <a
          class="superBtn"
          onClick={[Function]}
>
          Favorites:
          <span>
                     
                    1
          </span>
</a>,
        "className": "favorites",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Favorites:",
            <span>
               
              1
</span>,
          ],
          "class": "superBtn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "Favorites:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                1,
              ],
            },
            "ref": null,
            "rendered": Array [
              " ",
              1,
            ],
            "type": "span",
          },
        ],
        "type": "a",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
